src/serac/physics/utilities/functional/tests/functional_boundary_test.cpp

```diff
@@ -74,21 +130,38 @@ void boundary_test(mfem::ParMesh& mesh, H1<p> test, H1<p> trial, Dimension<dim>)
       [&](auto x, auto n, auto u) {
         tensor<double, dim> b{sin(x[0]), x[0] * x[1]};
         return x[0] * x[1] + dot(b, n) + rho * u;
-      },
-      mesh);
+      }, mesh);
 
   mfem::Vector r1 = (*J) * U + (*F);
   mfem::Vector r2 = residual(U);
 
+  mfem::SparseMatrix gradient1 = B.SpMat();
+  mfem::SparseMatrix gradient2 = grad(residual);
+
   if (verbose) {
     std::cout << "sum(r1):  " << r1.Sum() << std::endl;
     std::cout << "sum(r2):  " << r2.Sum() << std::endl;
     std::cout << "||r1||: " << r1.Norml2() << std::endl;
     std::cout << "||r2||: " << r2.Norml2() << std::endl;
     std::cout << "||r1-r2||/||r1||: " << mfem::Vector(r1 - r2).Norml2() / r1.Norml2() << std::endl;
+
+    std::ofstream outfile;
+
+    outfile.open(std::to_string(p) + "_" + std::to_string(dim) + "_A_mfem.mtx");
+    gradient1.SortColumnIndices();
+    gradient1.PrintMM(outfile);
```

Comment:
Does the matrix printing work in parallel?

Reply:
I'm not sure. Right now, the SparseMatrix that I form isn't a HypreParMatrix, so it's just printing the local processor's contribution